sklearn/metrics/_plot/base.py

```diff
@@ -1,116 +1,56 @@
-
-    pos_label: str or int
-        The class considered as the positive class when computing
-        the metrics.
+        An estimator that should be used to predict the target.
+
+    y : ndarray
+        The associated target.
+
+    target_type : str, default=None
+        The type of the target `y` as returned by
+        :func:`~sklearn.utils.multiclass.type_of_target`. If `None`, the type
+        will be inferred by calling :func:`~sklearn.utils.multiclass.type_of_target`.
+        Providing the type of the target could save time by avoid calling the
+        :func:`~sklearn.utils.multiclass.type_of_target` function.
+
+    Raises
+    ------
+    ValueError
+        If the estimator or the target are not binary.
     """
-    classification_error = (
-        "Expected 'estimator' to be a binary classifier, but got"
-        f" {estimator.__class__.__name__}"
-    )
+    try:
+        check_is_fitted(estimator)
+    except NotFittedError as e:
+        raise NotFittedError(
+            f"This {estimator.__class__.__name__} instance is not fitted yet. Call "
+            "'fit' with appropriate arguments before intending to use it to plotting "
+            "functionalities."
+        ) from e
 
     if not is_classifier(estimator):
-        raise ValueError(classification_error)
-
-    prediction_method = _check_classifier_response_method(estimator, response_method)
-    y_pred = prediction_method(X)
-    if pos_label is not None:
-        try:
-            class_idx = estimator.classes_.tolist().index(pos_label)
-        except ValueError as e:
-            raise ValueError(
-                "The class provided by 'pos_label' is unknown. Got "
-                f"{pos_label} instead of one of {set(estimator.classes_)}"
-            ) from e
-    else:
-        class_idx = 1
-        pos_label = estimator.classes_[class_idx]
-
-    if y_pred.ndim != 1:  # `predict_proba`
-        y_pred_shape = y_pred.shape[1]
-        if y_pred_shape != 2:
-            raise ValueError(
-                f"{classification_error} fit on multiclass ({y_pred_shape} classes)"
-                " data"
-            )
-        y_pred = y_pred[:, class_idx]
-    elif pos_label == estimator.classes_[0]:  # `decision_function`
-        y_pred *= -1
+        raise ValueError(
+            "This plotting functionalities only support a binary classifier. "
+            f"Got a {estimator.__class__.__name__} instead."
+        )
+    elif len(estimator.classes_) != 2:
+        raise ValueError(
+            f"This {estimator.__class__.__name__} instance is not a binary "
+            "classifier. It was fitted on multiclass problem with "
+            f"{len(estimator.classes_)} classes."
+        )
 
-    return y_pred, pos_label
+    if target_type is None:
+        target_type = type_of_target(y)
+    if target_type != "binary":
+        raise ValueError(
+            f"The target y is not binary. Got {target_type} type of target."
+        )
```

Comment:
I think all of the metric functions, `det_curve`, `calibration_cruve`, etc already checks for binary `y_true`. Do we need to check `y_true` here?